ENT-9380: Added doc for policy function filereadable

larsewi opened the pull request:
Depends on: https://github.com/cfengine/core/pull/5147

larsewi committed:
Added doc for policy function isreadable

Ticket: ENT-9380
Changelog: Title
Signed-off-by: Lars Erik Wik <lars.erik.wik@northern.tech>
Co-authored-by: Ole Herman Schumacher Elgesem <olehermanse@users.noreply.github.com>

larsewi commented:
[![Build Status](https://ci.cfengine.com/buildStatus/icon?job=pr-pipeline&build=9130)](https://ci.cfengine.com/job/pr-pipeline/9130/)

larsewi commented:
@cf-bottom are you back from vacation now? If so, can I have a build in Jenkins? :)

cf-bottom commented:
Alright, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=build-and-deploy-docs-master&build=69)](https://ci.cfengine.com//job/build-and-deploy-docs-master/69/)

**Jenkins:** https://ci.cfengine.com/job/build-and-deploy-docs-master/69/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-build-and-deploy-docs-master-69/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-build-and-deploy-docs-master-69/output/_site/

nickanderson commented:
Pushed a couple small changes directly to your branch @larsewi.

@cf-bottom jenkins, please

cf-bottom commented:
Alright, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=build-and-deploy-docs-master&build=71)](https://ci.cfengine.com//job/build-and-deploy-docs-master/71/)

**Jenkins:** https://ci.cfengine.com/job/build-and-deploy-docs-master/71/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-build-and-deploy-docs-master-71/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-build-and-deploy-docs-master-71/output/_site/

nickanderson commented on reference/functions/filereadable.markdown:
```diff
@@ -0,0 +1,40 @@
+
+[%CFEngine_include_snippet(filereadable.cf, #\+begin_src\s+example_output\s*, .*end_src)%]
+
+**History:** Introduced in 3.21.
```
```suggestion
**History:** Introduced in 3.22.0
```

nickanderson reviewed (commented):
3.21 has come and gone, so this will have to be 3.22.0 at least.

larsewi commented:
@nickanderson, I think I accidentally removed your changes by force 😬 Sorry! Can't seem to find the commit in the "conversation" tab. Not sure how to fix this.

nickanderson commented:
![image](https://user-images.githubusercontent.com/202896/212125585-633b0446-936d-4944-83b2-b33bbb270d13.png)
![image](https://user-images.githubusercontent.com/202896/212125646-ea6075dd-3986-44a8-9c9c-8753ab3602bb.png)

larsewi commented:
@nickanderson did you not add some functions to see also?

larsewi commented:
@cf-bottom Jenkins :)

cf-bottom commented:
Alright, I triggered a build:

[![Build Status](https://ci.cfengine.com//buildStatus/icon?job=build-and-deploy-docs-master&build=74)](https://ci.cfengine.com//job/build-and-deploy-docs-master/74/)

**Jenkins:** https://ci.cfengine.com/job/build-and-deploy-docs-master/74/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-build-and-deploy-docs-master-74/

**Documentation:** http://buildcache.cfengine.com/packages/build-documentation-pr/jenkins-build-and-deploy-docs-master-74/output/_site/

nickanderson commented:
> @nickanderson did you not add some functions to see also?

Not that I recall. But `isexecutable()`, `isdir()`, `islink()`, `isplain()` come to mind right now. Looking at those function names, makes me slightly question your name choice.

larsewi commented:
> > @nickanderson did you not add some functions to see also?

> Not that I recall. But isexecutable(), isdir(), islink(), isplain() come to mind right now. Looking at those function names, makes me slightly question your name choice.

Changed name and added the "see also" functions

olehermanse commented on reference/functions/isreadable.markdown:
```diff
@@ -0,0 +1,45 @@
+
+[%CFEngine_function_prototype(path, timeout)%]
+
+**Description:** Check if file is readable.
```
```suggestion
**Description:** Check if a file is readable.
```

olehermanse commented on reference/functions/isreadable.markdown:
```diff
@@ -0,0 +1,45 @@
+
+**Description:** Check if file is readable.
+
+This function checks if file specified by the `path` parameter is readable by
```
```suggestion
This function checks if the file specified by the `path` parameter is readable by
```

olehermanse commented on reference/functions/isreadable.markdown:
```diff
@@ -0,0 +1,45 @@
+**Description:** Check if file is readable.
+
+This function checks if file specified by the `path` parameter is readable by
+trying to read 1 Byte from it. If file is not ready for I/O, and the `timeout`
```
```suggestion
trying to read 1 Byte from it. If the file is not ready for I/O, and the `timeout`
```

olehermanse commented on reference/functions/isreadable.markdown:
```diff
@@ -0,0 +1,45 @@
+
+This function checks if file specified by the `path` parameter is readable by
+trying to read 1 Byte from it. If file is not ready for I/O, and the `timeout`
+parameter is not set to 0, the read operation will block for at most N number of
+seconds specified by the `timeout` parameter.
```
```suggestion
parameter is not set to 0, the read operation will block for at most `timeout` number of seconds.
```

olehermanse commented on reference/functions/isreadable.markdown:
```diff
@@ -0,0 +1,45 @@
+parameter is not set to 0, the read operation will block for at most N number of
+seconds specified by the `timeout` parameter.
+
+The `timeout` parameter is optional and defaults to 3 seconds if it is not
```
```suggestion
The `timeout` parameter is optional and defaults to 3 seconds, if it is not
```

olehermanse commented on reference/functions/isreadable.markdown:
```diff
@@ -0,0 +1,45 @@
+indefinetly.
+
+The function evaluates to false if the time limit expires or the read operation
+failes for any other reason (e.g. permission denied).
```
```suggestion
fails for any other reason (e.g. permission denied).
```

olehermanse commented on reference/functions/isreadable.markdown:
```diff
@@ -0,0 +1,45 @@
+
+When the `timeout` parameter is not 0, the agent will spawn a separate thread
+that will try to open and read the file. The agent will wait for at most N
+number of seconds for the spawned thread to finnish. If the thread does not
```
```suggestion
number of seconds for the spawned thread to finish. If the thread does not
```

olehermanse commented on reference/functions/isreadable.markdown:
```diff
@@ -0,0 +1,45 @@
+When the `timeout` parameter is not 0, the agent will spawn a separate thread
+that will try to open and read the file. The agent will wait for at most N
+number of seconds for the spawned thread to finnish. If the thread does not
+finnish in time, the agent will cancel the thread execution, and concider the
```
```suggestion
finish in time, the agent will cancel the thread execution, and consider the
```

nickanderson replied:
got something against the fins? What if  it was sweedish in time?

olehermanse commented on reference/functions/isreadable.markdown:
```diff
@@ -0,0 +1,45 @@
+number of seconds for the spawned thread to finnish. If the thread does not
+finnish in time, the agent will cancel the thread execution, and concider the
+file unreadable.
+
```
```suggestion

If the file is of size 0, the function will return true, if it successfully reads 0 bytes (reaches end-of-file).

```
